java/client/src/main/java/glide/api/commands/HashCommands.java

```diff
@@ -45,4 +45,20 @@ public interface HashCommands {
      *     If <code>key</code> does not exist, it is treated as an empty hash and it returns 0.<br>
      */
     CompletableFuture<Long> hdel(String key, String[] fields);
+
+    /**
+     * Returns all fields and values of the hash stored at <code>key</code>.
+     *
+     * @see <a href="https://redis.io/commands/hgetall/">redis.io</a> for details.
+     * @param key The key of the hash.
+     * @return A <code>Map</code> of fields and their values stored in the hash. Every field name in
+     *     the list is followed by its value. <br>
+     *     If <code>key</code> does not exist, it returns an empty map.
+     * @example
+     *     <pre>
+     * Map fieldValueMap = client.hgetall("my_hash").get()
+     * assert fieldValueMap == Map.of(field1", "value1", "field2", "value2")
```

Comment:
```suggestion
     * assert fieldValueMap.equals(Map.of(field1", "value1", "field2", "value2"))
```
operator `==` sucks in java

Reply:
just recalled that you have array comparioson with `==` somewhere...